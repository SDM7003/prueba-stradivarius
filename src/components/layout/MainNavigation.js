import {Link} from "react-router-dom";

import classes from "./MainNavigation.module.css";
import $ from 'jquery';
import {useFetch} from "../../util-hooks/useFetch";

export default function MainNavigation({ setPage }) {
  $(document).ready(function () {
    var lastScrollTop = 0;
    $(window).scroll(function (event) {
      var st = $(this).scrollTop();
      if (st < lastScrollTop) {
        $('header').slideDown();
      } else {
        $('header').slideUp();
      }
      lastScrollTop = st;
    });
  });

  const { data } = useFetch({
    url: "/data.json",
  });

  if (!data) return <p>Loading...</p>;

  let [fav] = data;

  let counter=0;
  for (const fav of data){
    if (fav.important === true){
      counter++;
    }
    console.log(counter);
  }


  return (
    <header className={classes.header} data-test="navigation-header">
      <div className={classes.logo}>React Meetups</div>
        <nav>
          <ul>
            <li>
              <Link to="/">
                All Meetups
              </Link>
            </li>

            <li>
              <Link to="/newMeetUp">
                Add New Meetup
              </Link>
            </li>
            <li>
              <Link to="/favorites">
                My Favorites
                <span className={classes.badge}>{counter}</span>
              </Link>
            </li>
          </ul>
        </nav>
    </header>
  );
}

