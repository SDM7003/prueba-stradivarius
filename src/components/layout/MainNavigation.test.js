/* eslint-disable testing-library/no-debugging-utils */
import { shallow } from "enzyme";
import MainNavigation from "./MainNavigation";
import {render} from "@testing-library/react";

test("<MainNavigation/> renders without crashing", () => {
  const wrapper = shallow(<MainNavigation/>);
  expect(wrapper.exists()).toBe(true);
});
test("render a snapshot",()=>{
  const {baseElem}=render(<MainNavigation/>);
  expect(baseElem).toMatchSnapshot();
})
