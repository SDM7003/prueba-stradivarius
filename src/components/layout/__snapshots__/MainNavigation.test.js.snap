// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a snapshot 1`] = `undefined`;
